File: practice/algorithms/strings/super_reduced_string/super_reduced_string.rb
s = gets.strip
acc = ""
for i in 0...(s.length) do
  c = s[i]
  if(acc.length > 0 && acc[acc.length - 1] == c)
    acc = acc.slice(0, acc.length - 1)
  else
    acc = acc + c
  end
end

if(acc.length == 0)
  puts "Empty String"
else
  puts acc
end

File: practice/algorithms/strings/super_reduced_string/SuperReducedString.js
'use strict';

const processData = input => {
    const line = input.split('\n')[0];
    let acc = "";
    for(let i = 0; i < line.length; i++) {
        const c = line[i];
        if(acc.length > 0 && acc[acc.length - 1] === c) {
            acc = acc.slice(0, acc.length - 1);
        } else {
            acc += c;
        }
    }
    console.log(acc.length === 0 ? "Empty String" : acc);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));